corgi/core/models.py

```diff
@@ -1780,6 +1831,7 @@ def cpes(self) -> QuerySet:
         """Build and return a list of CPEs from all Variants this Component relates to"""
         # For each Variant-type relation, get the linked Variant's CPE directly
         # Remove any duplicates, and return the CPEs in sorted order so manifests are stable
+        # TODO review this now that we have INFERRED variants
         return (
             self.productvariants.exclude(cpe="")
```

Comment:
I think you can just filter a little more and still get the correct results:
```suggestion
            self.productvariants.filter(pnodes__type=ProductNode.ProductNodeType.DIRECT).exclude(cpe="")
```